### ISSUE TITLE
Option default must be a kind of [String, Array, Hash]!  You passed false.

### ISSUE THREAD

**Issue by sstelfox** [opscode-cookbooks/build-essential#56]
The recent PR that got merged five days ago resulted in the following error starting to show up when we try to make use of build-essential:

    Chef::Exceptions::ValidationFailed: Option default must be a kind of [String, Array, Hash]!  You passed false.

I'd make a PR but I'm not sure what the correct value should be.

**Comment by STRML** [comment]
Seeing the same on opsworks.

**Comment by g3kk0** [comment]
+1

**Comment by chrisking** [comment]
+1 as well

**Comment by chrisking** [comment]
I believe the error was actually introduce via the metadata.rb file but not entirely sure: https://github.com/opscode-cookbooks/build-essential/commit/b4785e95223867eec6cd2b6a87d22818bd5453fb#diff-9fd042d4cd1e8f82894bbd7dffb313ec

**Comment by Beagle25** [comment]
For anyone trying to get around this, changing the boolean false on line 28 of metadata.rb to a string, i.e. "false" worked for me. Good luck.

**Comment by keen99** [comment]
we ran into this the other day and hacked around it - but currently build-essentials is broken for chef before 11.12.10.   

I discussed it with flaccid, who authored the original patch a long time ago, and he dug for where support for what he patched came from.   The original breaking commit for build-essential:

https://github.com/opscode-cookbooks/build-essential/commit/75793b142082836e75a88764dd9a652dfa30db63


the change in chef that added support for bools was here:

https://github.com/chef/chef/commit/ee236d095f8d2e2a0d3cb0e158066fd2b654a54d#diff-e7f90eec970c3a67a738e9000abf630cL394

which made it into release here:

https://github.com/chef/chef/blob/11.12.0/lib/chef/cookbook/metadata.rb#L394


switching this to a string, instead of bool, as suggested by @Beagle25 seems to be the appropriate fix.

that said, this raises the question of why we're not testing this cookbook against $all versions of chef before releasing patches?

**Comment by keen99** [comment]
@flaccid - any chance you can confirm that using a string instead of a bool here would work for the need you originally patched this for? :)

**Comment by flaccid** [comment]
https://github.com/opscode-cookbooks/build-essential/blob/master/libraries/timing.rb#L62 has double bang, but I'm not sure its what we think. Would appreciate another pair of ruby eyes.
```
irb(main):014:0> compile_time = false
=> false
irb(main):015:0> !!compile_time
=> false
irb(main):016:0> compile_time = 'false'
=> "false"
irb(main):017:0> !!compile_time
=> true
```

**Comment by jtimberman** [comment]
Is the issue in question here regarding the metadata `attribute 'build-essential/compile_time'` having `:default => false`? When does this manifest itself? What command are you running where this occurs? Yes indeed, the capability was added to cookbook metadata to have boolean values, but it still remains true that other than user interfaces that consume that data, nothing inside of Chef itself makes use of the attributes in metadata.

I'm :+1: on #57 to address this, but I strongly encourage folks to upgrade to a newer version of Chef - 11.12.0 itself was released April 8, 2014, and the previous version, 11.10.4 was released on February 20, 2014. It's not sustainable for us to update every cookbook to ensure that every old version of Chef that doesn't have new features that are introduced in later versions of Chef. What is the purpose in having new versions with new features if we're not using them too? 

For those who can't keep their Chef version updated to the current release, I then encourage you to pin your cookbook versions to older releases that don't have the new features that don't work on your versions of Chef.

**Comment by flaccid** [comment]
Its a bit of a vicious circle where installing the packages at compile time is needed but the chef version used is static on opsworks or rightscale. I presume a newer version of Chef that is sufficient will come out not too far into the future for RightScale RightLink. The workaround just being to fork it and make it a string for now.

**Comment by schisamo** [comment]
This was fixed in: dfcf0b34d3581387f590f39fba5f312a85c3ad22
